lib/msf/core/post/file.rb

```diff
@@ -362,10 +362,23 @@ def read_file(file_name)
     session.shell_command_token("while read line; do echo $line; done <#{file_name}")
   end
 
+  # Checks to see if there are non-ansi or newline characters in a given string
+  #
+  # @param data [String] String to check for non-ansi or newline chars
+  # @return bool
+  def is_safe?(data)
+    index = 0
+    data.each_char do |char|
+      index += 1
+      unless char.ascii_only? or char == '\n'
+        return false
+      end
+    end
+    return true
+  end
+
   # Platform-agnostic file write. Writes given object content to a remote file.
   #
-  # NOTE: *This is not binary-safe on Windows shell sessions!*
-  #
   # @param file_name [String] Remote file name to write
   # @param data [String] Contents to put in the file
   # @return [void]
```

Comment:
In the future we could potentially expose the chunk size to the user:
```
  def write_file(file_name, data, chunk_size: 5000)
```

Reply:
I think I might have misunderstood your comments on this; as suggested, a we could not alter the 5000 chunk size without a code change, which does not seem to me like the right path.